gateway/gateway-runtime/policy-engine/cmd/policy-engine/main_test.go

```
@@ -331,3 +335,116 @@ func TestInitializeXDSClient_ValidConfig(t *testing.T) {
+func (failingWriter) Write([]byte) (int, error) {
+	return 0, errors.New("write failed")
+}
+
+// captureStdout redirects os.Stdout for the duration of fn. setupLogger binds to
+// os.Stdout at construction, so the swap must precede the call.
+func captureStdout(t *testing.T, fn func()) string {
+	t.Helper()
+
+	r, w, err := os.Pipe()
+	require.NoError(t, err)
+	defer r.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `r.Close` is not checked

(errcheck)